### ISSUE TITLE
Check for numeric variables

### ISSUE THREAD

**Issue by bjornerstedt** [tidymodels/corrr#138]
## Feature: drop non-numeric variables

As correlation coefficients are not meaningful for factor variables and strings, it would be convenient if they were simply dropped. This is standard behaviour in Stata for example; variables are dropped with a warning message. 

Dropping variables before invoking `correlate` is of course relatively easy to do. One can for instance use dplyr;

```r
mpg %>% select_if( is.numeric) %>% correlate(quiet = TRUE)
```

I still believe that the behaviour should be changed for several reasons. 

* Dropping variables is more convenient than receiving an error message. Convenient defaults matter especially for new users in teaching environments. 
* Factor and string variables are common in datasets. 
* Dropping variables rather than giving an error is a common practice in Stata and R (for example dropping perfectly multicollinear variables from an estimate rather than throwing an error). 
* Modifying the behaviour should not break existing code.

Alternatively, corrr could provide a convenience function `correlate_all` that provides this feature.

**Comment by juliasilge** [comment]
I know that @thisisdaryn has thought a lot about this issue and has worked up a solution that is close to being ready to go. Thoughts?

**Comment by thisisdaryn** [comment]
Yes, I agree that `correlate()` should drop non-numeric variables by default. I have to add some tests to my PR.

**Comment by juliasilge** [comment]
Closed in #139 


The `correlate()` function will now drop non-numeric variables with a message:

``` r
library(corrr)
correlate(dplyr::storms)
#> Non-numeric variables removed from input: `name`, `status`, and `category`
#> Correlation computed with
#> * Method: 'pearson'
#> * Missing treated using: 'pairwise.complete.obs'
#> # A tibble: 10 x 11
#>    term      year    month      day     hour      lat     long     wind pressure
#>    <chr>    <dbl>    <dbl>    <dbl>    <dbl>    <dbl>    <dbl>    <dbl>    <dbl>
#>  1 year  NA       -0.0115   1.84e-2  1.57e-3 -0.121    0.0604   0.0490  -0.0726 
#>  2 month -0.0115  NA       -1.83e-1 -5.12e-3 -0.0659   0.0484   0.127   -0.134  
#>  3 day    0.0184  -0.183   NA        7.16e-4 -0.0509   0.0406  -0.00650 -0.00101
#>  4 hour   0.00157 -0.00512  7.16e-4 NA        0.00268 -0.00919  0.00183  0.00160
#>  5 lat   -0.121   -0.0659  -5.09e-2  2.68e-3 NA       -0.104    0.0761  -0.104  
#>  6 long   0.0604   0.0484   4.06e-2 -9.19e-3 -0.104   NA        0.00474  0.0585 
#>  7 wind   0.0490   0.127   -6.50e-3  1.83e-3  0.0761   0.00474 NA       -0.942  
#>  8 pres… -0.0726  -0.134   -1.01e-3  1.60e-3 -0.104    0.0585  -0.942   NA      
#>  9 ts_d…  0.0212   0.139    2.01e-2  8.56e-3  0.301   -0.0146   0.640   -0.683  
#> 10 hu_d… -0.0997   0.112    3.39e-2  5.05e-3  0.164   -0.102    0.774   -0.842  
#> # … with 2 more variables: ts_diameter <dbl>, hu_diameter <dbl>
```

<sup>Created on 2021-07-06 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>
